sdk/src/keystore/mod.rs

```diff
@@ -44,14 +44,133 @@ pub mod error;
 /// Schnorrkel Support
 pub mod sr25519;
 
-use alloc::string::ToString;
+use crate::clients::tangle::runtime::TangleConfig;
+use crate::keystore::sp_core_subxt::crypto::{DeriveError, SecretStringError};
+use crate::keystore::sp_core_subxt::DeriveJunction;
+#[cfg(any(feature = "std", feature = "wasm"))]
+// TODO: Once subxt uses sp-core 34.0.0, we can simply use sp_core
+pub use crate::tangle_subxt::subxt::ext::sp_core as sp_core_subxt;
+use alloy_signer_local::LocalSigner;
 use eigensdk_rs::eigen_utils::crypto::bls::{self as bls_bn254, g1_point_to_g1_projective};
 pub use error::Error;
-use schnorrkel::SignatureError;
-use subxt_core::config::PolkadotConfig;
-use subxt_core::tx::signer::PairSigner;
+use k256::ecdsa::SigningKey;
+use subxt::ext::sp_core::Pair as PairSubxt;
+use subxt_core::tx::signer::{PairSigner, Signer};
+use subxt_core::utils::{AccountId32, MultiAddress, MultiSignature};
+#[cfg(any(feature = "std", feature = "wasm"))]
+use tangle_subxt::subxt;
 
-pub type TanglePairSigner = PairSigner<PolkadotConfig, subxt_core::ext::sp_core::sr25519::Pair>;
+#[cfg(any(feature = "std", feature = "wasm"))]
+#[derive(Clone, Debug)]
+pub struct TanglePairSigner<Pair> {
+    pub(crate) pair: subxt::tx::PairSigner<TangleConfig, Pair>,
+}
+
+#[cfg(any(feature = "std", feature = "wasm"))]
+pub type TangleSr25519PairSigner = TanglePairSigner<sp_core_subxt::sr25519::Pair>;
```

Comment:
Theres a mix of `TangleSr25519PairSigner` and `TanglePairSigner<ecdsa::Pair>`. Why not make an alias for ECDSA as well? Or just use no aliases.

Reply:
We could, though, I'd have to make an alias for the others as well. Since `TangleSr25519PairSigner` is used much more than the related ecdsa pair, I make it a special case of having its own type.

Reply:
If my PR gets through let's not use aliases. It's more and more unnecessary code that. Alex is right. We should be consistent with what we are going across the repo or we should not do something at all if its not needed.